test/h5test.c

```diff
@@ -2008,16 +2008,20 @@ H5_get_srcdir_filename(const char *filename)
 
     /* Check for error */
     if (NULL == srcdir)
-        return (NULL);
-    else {
-        /* Build path to test file */
-        if ((HDstrlen(srcdir) + HDstrlen(filename) + 1) < sizeof(srcdir_testpath)) {
-            HDsnprintf(srcdir_testpath, sizeof(srcdir_testpath), "%s%s", srcdir, filename);
-            return (srcdir_testpath);
-        } /* end if */
-        else
-            return (NULL);
-    } /* end else */
+        return NULL;
+
+    /* Build path to test file. We're checking the length so suppress
+     * the gcc format-truncation warning.
+     */
+    if ((HDstrlen(srcdir) + HDstrlen(filename) + 1) < sizeof(srcdir_testpath)) {
+        H5_GCC_DIAG_OFF("format-truncation")
+        HDsnprintf(srcdir_testpath, sizeof(srcdir_testpath), "%s%s", srcdir, filename);
```

Comment:
This one should be easily fixable without needing to suppress the warning

Reply:
Agree